generate_analysis_graphs.py

```diff
@@ -84,6 +87,53 @@
         individuals = TracedDataJsonIO.import_jsonl_to_traced_data_iterable(f)
     log.info(f"Loaded {len(individuals)} individuals")
 
+    # Compute the number of messages, activations and participants per episode and overall.
+    log.info("Computing the per-episode and per-season engagement counts...")
+    engagement_counts = OrderedDict()
+    for plan in PipelineConfiguration.RQA_CODING_PLANS:
+        engagement_counts[plan.dataset_name] = {
+            "Episode": plan.dataset_name,
+            "Total messages with Opt-ins": 0,
+            "Total participants per-show with Opt-ins": 0,
+            "Total participants with Opt-ins": '-'
+        }
+    engagement_counts["Total"] = {
+        "Episode": "Total",
+        "Total messages with Opt-ins": 0,
+        "Total participants per-show with Opt-ins": '-',
+        "Total participants with Opt-ins": 0,
+    }
+
+    # Compute, per episode and across the season:
+    #  - Total Messages, by counting the number of consenting message objects that contain the raw_field key each week.
+    for msg in messages:
+        if msg["consent_withdrawn"] == Codes.FALSE:
+            for plan in PipelineConfiguration.RQA_CODING_PLANS:
+                if plan.raw_field in msg:
+                    engagement_counts[plan.dataset_name]["Total messages with Opt-ins"] += 1
+                    engagement_counts["Total"]["Total messages with Opt-ins"] += 1
+
+    # Compute, per episode and across the season:
+    #  - Total Participants, by counting the number of consenting individuals objects that contain the raw_field key
+    #    each week.
+    for ind in individuals:
+        if ind["consent_withdrawn"] == Codes.FALSE:
+            engagement_counts["Total"]["Total participants with Opt-ins"] += 1
+            for plan in PipelineConfiguration.RQA_CODING_PLANS:
+                if plan.raw_field in ind:
+                    engagement_counts[plan.dataset_name]["Total participants per-show with Opt-ins"] += 1
+
+    # Export the engagement counts to a csv.
+    with open(f"{output_dir}/engagement_counts.csv", "w") as f:
+        headers = ["Episode", "Total messages with Opt-ins",
+                   "Total participants per-show with Opt-ins", "Total participants with Opt-ins"]
```

Comment:
I think "participants per-show" is now called activations?

Reply:
activations only apply to participation across all episodes but not within one episode.

Reply:
@lukechurch